Util.py

```diff
@@ -2183,25 +2183,35 @@ def unicode_to_str(s):
   return s
 
 
-def deepcopy(x):
+def deepcopy(x, stop_types=None):
   """
   Simpler variant of copy.deepcopy().
   Should handle some edge cases as well, like copying module references.
 
   :param T x: an arbitrary object
+  :param list[type]|None stop_types: objects of these types will not be deep-copied, only the reference is passed
   :rtype: T
   """
   # See also class Pickler from TaskSystem.
   # Or: https://mail.python.org/pipermail/python-ideas/2013-July/021959.html
   from TaskSystem import Pickler, Unpickler
 
+  persistent_memo = {}
```

Comment:
I think it would be good to add a comment of the content:
```suggestion
  persistent_memo = {}  # id -> obj
```

Reply:
Do you want things like this as user hint only, or as PyCharm type hint, such as `persistent_memo = {}  # type: dict[id, object]`?

Reply:
(There should be two spaces.)
This is not a valid PyCharm type hint. The valid PyCharm type hint would be `typing.Dict[int,object]`.
However, even more important here is the semantic information, `id -> obj`. The type is probably already correctly automatically derived by PyCharm. I leave it to you if you want to have both here or just the semantic info.

Reply:
For me this was working in PyCharm, but you are right that typing should be used. I will just use short info.

Reply:
This definitely does not work. It probably ignored it (but even with a warning that it is incorrect), and just used the auto-derived type.

Reply:
It definitely did change the function autocompletion suggestion when I edited the types in the hint.

Reply:
Please try it for yourself in an empty file, for me there is no warning, and correct autocompletion hints.